Src/Particle/AMReX_ParticleContainerBase.H

```diff
@@ -245,6 +245,14 @@ public:
     template <class MF>
     bool OnSameGrids (int level, const MF& mf) const { return m_gdb->OnSameGrids(level, mf); }
 
+    [[nodiscard]] Arena* arena () const {
+        return m_arena;
+    }
+
+    void SetArena (Arena* a) {
+        m_arena = a;
+    }
```

Comment:
@AlexanderSinn @atmyers I am wondering: should we add an assert here that setting the arena can only be done once?

There are already good asserts / exceptions now if a PC with an unset allocator is trying to add particles. Thus, the logic here could be as easy as: assert if `m_arena` is not `nullptr` when callint `SetArena`. What do you think?

I want to avoid that someone calls `SetArena` for an already defined PC with live particles and gets into an unexpected state by that.

Reply:
There is the `a_arena == GetStructOfArrays().GetIdCPUData().arena()` check in ParticleTile that should catch this issue, but I guess we could add another check to ParticleContainerBase 